plugins/catalog-backend/src/database/DefaultProcessingDatabase.ts

```diff
@@ -516,6 +532,31 @@ export class DefaultProcessingDatabase implements ProcessingDatabase {
     }
   }
 
+  async refreshByRefreshKeys(
+    txOpaque: Transaction,
+    options: RefreshByKeyOptions,
+  ) {
+    const tx = txOpaque as Knex.Transaction;
+    const { keys } = options;
+
+    const updateResult = await tx<DbRefreshStateRow>('refresh_state')
+      .whereIn('entity_ref', function selectEntityRefs(tx2) {
+        tx2
+          .whereIn('key', keys)
```

Comment:
It's probably worth doing a lil bit of manual batching here tbh, chunking the keys up by however many values `min(SQLite, PostgreSQL)` supports for `IN`. Could easily have a PR that changes thousands of files and have this blow up right?

Reply:
Based on some google searches it seems that postgres doesn't have direct limit for the IN condition, the implicit limit is 32767. For SQLite there aren't explicit limit for the IN condition either, but the maximum_length_of a query is: 1000000

In this case you still think we should do some manual batching ? It seem the in condition should be able to handle some pretty big arrays.